### PR TITLE
feat(gradle): add *gradle.lockfile parser

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add [*gradle.lockfile](https://docs.gradle.org/current/userguide/dependency_locking.html#lock_state_location_and_format) parser.

## Related issue:
- aquasecurity/trivy/issues/2707

### CODE REVIEW

File: pkg/gradle/lockfile/parse.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var libs []types.Library
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") { // skip comments
```
Comment (afdesk) [inline pkg/gradle/lockfile/parse.go:23]: I'd use something like [TrimSpace](https://pkg.go.dev/strings#TrimSpace) here, to skip next comments:
```yaml
 # comment
   # comment2
````

File: pkg/gradle/lockfile/parse.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+			continue
+		}
+
+		// dependency format: group:artifact:version=classPaths
+		dep := strings.Split(line, ":")
+		if len(dep) != 3 { // skip the last line with lists of empty configurations
```
Comment (afdesk) [inline pkg/gradle/lockfile/parse.go:29]: is this check enough? i mean that users can have strange (incorrect) lockfiles: https://github.com/tadfisher/gradle2nix/blob/master/gradle/dependency-locks/settings-classpath.lockfile

or it doesn't matter, right?

File: pkg/gradle/lockfile/testdata/happy.lockfile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# This is a Gradle generated file for dependency locking.
+# Manual edits can break the build and are not advised.
+# This file is expected to be part of source control.
+cglib:cglib-nodep:2.1.2=testRuntimeClasspath,classpath
+org.springframework:spring-asm:3.1.3.RELEASE=classpath
```
Comment (afdesk) [inline pkg/gradle/lockfile/testdata/happy.lockfile:5]: I'd also add a record from the gradle sample:
```
org.springframework:spring-beans:5.0.5.RELEASE=compileClasspath, runtimeClasspath
```

File: pkg/gradle/lockfile/parse.go (hunk 1 above)
Comment (afdesk) [reply pkg/gradle/lockfile/parse.go:23]: or it doesn't matter here too...

**Review by afdesk — APPROVED** [review]

File: pkg/gradle/lockfile/parse.go (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/gradle/lockfile/parse.go:29]: dependencies from this file don't have `=classPaths`. The rest looks default.
I think we can stay this check. If  there are problems with this, we will add more checks.

File: pkg/gradle/lockfile/parse.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/gradle/lockfile/parse.go:23]: It is make sense.
I am not sure that is possible, but i added trim to be safe.

File: pkg/gradle/lockfile/testdata/happy.lockfile (hunk 3 above)
Comment (DmitriyLewen) [reply pkg/gradle/lockfile/testdata/happy.lockfile]: Done

**Pull request merged**